### PR TITLE
drivers/sx127x: add support for rx and tx switch pin

### PR DESCRIPTION
**Pull request by fjmolinas**
### Contribution description

This PR adds support for RX/TX antena switch for sx1727x. Most of the boards used so far have this fucntion covered by hardware. Some of them don't, eg [im880b](https://www.wireless-solutions.de/downloads/Radio-Modules/iM880B/General_Information/iM880B_Datasheet_V1_6.pdf). In some cases this switch is devided in to two, as the case of IM880b, 1 for tx and another for rx. In other [boards](https://os.mbed.com/media/uploads/GregCr/sx1272mb2xas_e364v01a_sch.pdf) there is a single pin that can be toogled (High when TX). 

Define SX127X_TX_SWITCH, and if needed SX127X_RX_SWITCH in the boards makefiles.

### Testing procedure

Compile on two lora capable boards, where one is im880b:

`make BOARD=im880b DRIVER=sx172x -C tests/driver_sx127x clean -j flash`

Listen on one and send on the other with SX127X_TX_SWITCH and SX127X_RX_SWITCH flags enabled for sx127x, re-flash without and notice SNR/RSSI difference.

> - INFO # {Payload: "hello" (6 bytes), RSSI: 221, SNR: 9, TOA: 1188}
> - INFO # {Payload: "hello" (6 bytes), RSSI: 212, SNR: 7, TOA: 1188}

### Issues/PRs references

Can use #11315 to test

### CODE REVIEW

**Comment by kYc0o** [comment]
AFAIK the boards/lobaro-lorabox is using an "in house" version of im880b. Would you like to check if this configuration also apply for such board?

**Comment by fjmolinas** [comment]
@kYc0o I would need to know the internal connections in the lobaro-lorabox, can't seem to find a datasheet or schematic with this info, maybe @leandrolanzieri has one?

**Comment by leandrolanzieri** [comment]
@fjmolinas I have the schematics, I will send them to you.

**Comment by fjmolinas** [comment]
@kYc0o Looking at the schematic provided by @leandrolanzieri the antenna is not switched between rx and tx mode, in both cases the antenna is directly connected. Although this is not a wrong design according to semtech recommendations separate paths for RX and TX is better (eg using a switch). Anyway its neither here or there, the important part is it can't be used for lobaro-lorabox.

**Comment by fjmolinas** [comment]
@kYc0o @aabadie ping!

File: drivers/include/sx127x.h [hunk 1]
```diff
@@ -216,6 +216,10 @@ typedef struct {
     gpio_t dio5_pin;                   /**< Interrupt line DIO5 (not used) */
 #ifdef SX127X_USE_DIO_MULTI
     gpio_t dio_multi_pin;              /**< Interrupt line for multiple IRQs */
+#endif
+#if defined(SX127X_TX_SWITCH) || defined(SX127X_RX_SWITCH)
```
Comment (aabadie) [inline drivers/include/sx127x.h]:
```suggestion
#if defined(SX127X_RX_SWITCH)
    gpio_t rx_switch_pin;              /**< Rx antenna switch */
#endif
#if defined(SX127X_TX_SWITCH)
    gpio_t tx_switch_pin;              /**< Tx antenna switch */
#endif
```

**Comment by fjmolinas** [comment]
@aabadie comments addressed!

File: drivers/include/sx127x.h [hunk 2]
```diff
@@ -216,6 +216,10 @@ typedef struct {
     gpio_t dio5_pin;                   /**< Interrupt line DIO5 (not used) */
 #ifdef SX127X_USE_DIO_MULTI
     gpio_t dio_multi_pin;              /**< Interrupt line for multiple IRQs */
+#endif
+#if defined(SX127X_USE_TX_SWITCH) || defined(SX127X_USE_RX_SWITCH)
```
Comment (aabadie) [inline drivers/include/sx127x.h]: Very minor: I would guard each `rx/tx_switch_pin` attribute of the struct with it's corresponding macro.

Comment (aabadie) [reply drivers/include/sx127x.h]: Ok, I should have thought twice before commenting. I think it's fine like this.

**Comment by aabadie** [comment]
Just tested this PR on b-l072z-lrwan1 and didn't noticed any regression. I also know that this works on i880b board so let's merge.

**Pull request merged**